docs/uml/data-access.puml

```diff
@@ -0,0 +1,91 @@
+@startuml Data Access
+
+package "Data Access Layer" {
+    interface IBaseRepository {
+        + GetAll(T entity) : Task<int>
+        + GetById(T entity) : Task<int>
+        + Create(T entity) : Task<int>
+        + Update(T entity) : Task<int>
+        + Delele(T entity) : Task<bool>
```

Comment:
in CRUD operations:

      + GetAll(T entity) : Task<int>
        + GetById(T entity) : Task<int>
        + Create(T entity) : Task<int>
        + Update(T entity) : Task<int>
        + Delele(T entity) : Task<bool>
        It is better to keep in mind that when we ask for a list it is not an int, it is a list, we can use IEnumerable or IList, for getById, we ask for an id, not an entity, the same for delete

 + GetAll() : Task<IEnumerable<T>>  
        + GetById(T id) : Task<T>
        + Create(T entity) : Task<int>
        + Update(T entity) : Task<int>
        + Delele(T id) : Task<bool>